source/connect/connection-targets.txt

```diff
@@ -0,0 +1,125 @@
+.. _cpp-connection-targets:
+
+==========================
+Choose a Connection Target
+==========================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: connection string, URI, server, settings, client, stable api
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use a connection string and ``mongocxx::client`` object
+to connect to different types of MongoDB deployments.
+
+.. include:: /includes/assumes-instance.rst
+
+Atlas
+-----
+
+To connect to a MongoDB deployment on Atlas, include the following elements
+in your connection string:
+
+- URL of your Atlas cluster
+- MongoDB username
+- MongoDB password
+
+Then, pass your connection string to the ``mongocxx::uri`` constructor, and use
+the ``mongocxx::uri`` object to construct a ``mongocxx::client`` object.
+
+When you connect to Atlas, we recommend using the {+stable-api+} client option to avoid
+breaking changes when Atlas upgrades to a new version of {+mdb-server+}.
+To learn more about the {+stable-api+} feature, see the :ref:`{+stable-api+} page
+<cpp-stable-api>`.
+
+The following code shows how to use the {+driver-short+} to connect to an Atlas cluster.
+The code also uses the ``server_api_opts`` option to specify a {+stable-api+} version.
+
+.. literalinclude:: /includes/connect.cpp
+   :copyable: true
+   :start-after: // start-atlas
+   :end-before: // end-atlas
+   :language: cpp
+
+.. tip::
+
+   Follow the :atlas:`Atlas driver connection guide </driver-connection>`
+   to retrieve your connection string.
+
+Local Deployments
+-----------------
+
+To connect to a local MongoDB deployment, use ``localhost`` as the hostname. By
+default, the ``mongod`` process runs on port 27017, though you can customize this for
+your deployment.
+
+The following code shows how to use the {+driver-short+} to connect to a local MongoDB
+deployment:
+
+.. literalinclude:: /includes/connect.cpp
+      :language: cpp
+      :copyable: true
+      :start-after: // start-client
+      :end-before: // end-client
+
+Replica Sets
+------------
+
+To connect to a replica set, specify the hostnames (or IP addresses) and 
+port numbers of the replica-set members in your connection string.
+
+If you aren't able to provide a full list of hosts in the replica set, you can 
+specify one or more of the hosts in the replica set and instruct the {+driver-short+} to 
+perform automatic discovery to find the others. To instruct the driver to perform
+automatic discovery, perform one of the following actions:
+
+- Specify the name of the replica set as the value of the ``replicaSet`` parameter.
+- Specify ``false`` as the value of the ``directConnection`` parameter.
```

Comment:
S: clarify what these parameters are passed to
```suggestion
- Specify the name of the replica set as the value of the ``replicaSet``connection string parameter.
- Specify ``false`` as the value of the ``directConnection`` connection string parameter.
```

Reply:
Also, I think "option" might better describe `replicaSet` and `directConnection` rather than "parameter"

Reply:
makes sense. i went with "parameter" because i think that's what it's [technically called in a URL](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL#parameters), but i can see the ambiguity

Reply:
```suggestion
- Specify the name of the replica set as the value of the ``replicaSet`` parameter in the connection string.
- Specify ``false`` as the value of the ``directConnection`` parameter in the connection string.
```